scripts/lib/loot.zs

```diff
@@ -68,4 +75,158 @@ function tweak(
+██╔══██╗██╔══██║██║     ██╔═██╗ ██╔═══╝ ██╔══██║██║     ██╔═██╗     ██╔══██║██╔══██║██║╚██╗██║██║     ██║  ██║██╔══╝  ██╔══██╗
+██████╔╝██║  ██║╚██████╗██║  ██╗██║     ██║  ██║╚██████╗██║  ██╗    ██║  ██║██║  ██║██║ ╚████║███████╗██████╔╝███████╗██║  ██║
+╚═════╝ ╚═╝  ╚═╝ ╚═════╝╚═╝  ╚═╝╚═╝     ╚═╝  ╚═╝ ╚═════╝╚═╝  ╚═╝    ╚═╝  ╚═╝╚═╝  ╚═╝╚═╝  ╚═══╝╚══════╝╚═════╝ ╚══════╝╚═╝  ╚═╝
+*/
+
+function addBackpackPool(tableName as string) as void{
+    loottweaker.LootTweaker.getTable(tableName).addPool('lootBackpack', 1.0f, 1.0f, 0.0f, 0.0f);
+}
+
+function addBackpackEmpty(tableName as string, weight as int = 1) as void{
+    loottweaker.LootTweaker.getTable(tableName).getPool('lootBackpack').addEmptyEntry(weight, 0, [], 'empty');
+}
+
+function addBackpackWithLoot(tableName as string, lootCommon as IData[], lootUncommon as IData[], lootRare as IData[], weight as int = 1) as void{
+    loottweaker.LootTweaker.getTable(tableName).getPool('lootBackpack').addItemEntry(
+        <travelersbackpack:travelers_backpack>,
+        weight,0,
+        [Functions.zenscript(function(input as IItemStack, rng as IRandom, context as LootContext) as IItemStack{
+            var dataTag = [] as IData;
+            var slots = [] as IData;
+            for i in 0 .. 48{
+                slots +=[i];
+            }
+
+            for i in 0 .. rng.nextInt(48){
+                var table as IData[] = [];
+                val number = 4.0 - Math.log(rng.nextInt(1,65)) / Math.log(4);
+                if(number > 3){table = lootRare;}
+                else if(number >2){table = lootUncommon;}
+                else { table = lootCommon;}
+                val x = slots[rng.nextInt(slots.length)];
+                slots = slots.deepUpdate([x],REMOVE);
+                val item = table[rng.nextInt(table.length)];
+                var dataToAdd as IData = {Slot: x, Count: rng.nextInt(item.tab[0],item.tab[1]), id: item.item.id};
+                if(!isNull(item.item.tag)) dataToAdd = dataToAdd + {tag: item.item.tag};
+                if(!isNull(item.item.Damage)) dataToAdd = dataToAdd + {Damage: item.item.Damage};
+                
+                dataTag = dataTag.deepUpdate([dataToAdd],APPEND);
+            }
+            return input.withTag({Items: dataTag});
+        })],
+        [],
+        'lootBackpack'
+    );
+}
+
+function addBackpackForestryWithLoot(bagType as IItemStack, tableName as string, lootCommon as IData[], lootUncommon as IData[], lootRare as IData[], weight as int = 1) as void{
+    loottweaker.LootTweaker.getTable(tableName).getPool('lootBackpack').addItemEntry(
+        bagType,
+        weight,0,
+        [Functions.zenscript(function(input as IItemStack, rng as IRandom, context as LootContext) as IItemStack{
+            var dataTag = [] as IData;
+            var slots = [] as IData;
+            if(bagType.name.endsWith('t2')){
```

Comment:
`IItemStack.name` unsafe way to check an item. This is never used in code. It is better to do this via `IItemStack.definition.id`